docs/reference/data-streams/modify-data-streams-api.asciidoc

```diff
@@ -0,0 +1,112 @@
+[[modify-data-streams-api]]
+=== Modify data streams API
+++++
+<titleabbrev>Modify data streams API</titleabbrev>
+++++
+
+Performs one or more data stream modification actions in a single atomic
+operation.
+
+////
+[source,console]
+----
+PUT /_index_template/template
+{
+  "index_patterns": ["my-logs*"],
+  "data_stream": { }
+}
+
+POST my-logs/_doc/
+{
+    "message" : "trying out DS modification",
+    "@timestamp": "2021-01-01"
+}
+
+POST my-logs/_rollover
+
+POST index-to-add/_doc/
+{
+    "description" : "this is the index to be added to a data stream",
+    "@timestamp": "2021-01-01"
+}
+----
+////
+
+[source,console]
+----
+POST _data_stream/_modify
+{
+  "actions": [
+    {
+      "remove_backing_index": {
+        "data_stream": "my-logs",
+        "index": ".ds-my-logs-2021.10.15-000001"
```

Comment:
@debadair, the specific name of the data stream's backing index here is causing one of the doc integration tests to fail. I believe we have other docs such as the "Use a data stream" page where the names of date-dependent backing indices are used, but I don't know how those were set up to be tested successfully.

Reply:
The example on the use a data stream page sets up a not-timestamped index and substitutes that index name for the timestamped one. That's not going to work for this call, though. If you swap the actions, you could remove the index you add and the test should work. I'd call it something other than "index-to-add" if you do that. The other option would be to skip the test.